zkevm-circuits/src/mpt_circuit/helpers.rs

```diff
@@ -1352,75 +1355,21 @@ impl<F: Field> MainRLPGadget<F> {
         Ok(rlp_witness)
     }
 
-    pub(crate) fn create_view(
-        &self,
-        meta: &mut VirtualCells<F>,
-        cb: &mut MPTConstraintBuilder<F>,
-        rot: usize,
-        item_type: RlpItemType,
-    ) -> RLPItemView<F> {
-        circuit!([meta, cb.base], {
-            let is_string = self.rlp.is_string_at(meta, rot);
-            let tag = self.tag.rot(meta, rot);
-            let max_len = self.max_len.rot(meta, rot);
-            let len = self.len.rot(meta, rot);
-
-            // Check the tag value
-            require!(tag => self.tag(item_type).expr());
-            // Check the is_string value
-            if item_type == RlpItemType::Value || item_type == RlpItemType::Key {
-                require!(is_string => true);
-            }
-            // Hashes always have length 32
-            if item_type == RlpItemType::Hash {
-                require!(len => HASH_WIDTH);
-            }
-            if item_type == RlpItemType::Node {
-                // Nodes always have length 0 or 32 when a string, or are < 32 when a list
-                ifx! {is_string => {
-                    require!(max_len => self.max_length(item_type).expr());
-                    require!(len => [0, HASH_WIDTH]);
-                } elsex {
-                    require!(max_len => HASH_WIDTH - 1);
-                }}
-            } else {
-                require!(max_len => self.max_length(item_type).expr());
-            }
-        });
-        RLPItemView {
-            num_bytes: Some(self.num_bytes.rot(meta, rot)),
-            len: Some(self.len.rot(meta, rot)),
-            mult_diff: Some(self.mult_diff.rot(meta, rot)),
-            rlc_content: Some(self.rlc_content.rot(meta, rot)),
-            rlc_rlp: Some(self.rlc_rlp.rot(meta, rot)),
-            bytes: self.bytes.iter().map(|byte| byte.rot(meta, rot)).collect(),
-            is_short: Some(self.rlp.value.is_short.rot(meta, rot)),
-            is_long: Some(self.rlp.value.is_long.rot(meta, rot)),
-        }
-    }
-
     fn tag(&self, item_type: RlpItemType) -> FixedTableTag {
         if item_type == RlpItemType::Nibbles {
             FixedTableTag::RangeKeyLen16
         } else {
             FixedTableTag::RangeKeyLen256
         }
     }
-
-    fn max_length(&self, item_type: RlpItemType) -> usize {
-        match item_type {
-            RlpItemType::Node => 32,
-            RlpItemType::Value => 32,
-            RlpItemType::Hash => 32,
-            RlpItemType::Key => 33,
-            RlpItemType::Nibbles => 32,
-        }
-    }
 }
 
 /// Main RLP item
 #[derive(Clone, Debug, Default)]
 pub struct RLPItemView<F> {
+    below_limit: LtGadget<F, 1>,
+    leading_non_zero: LtGadget<F, 1>,
```

Comment:
Could use `IsNotZero` or `IsEqual` instead?

Reply:
Yeah that should do the work too :)